src/main/java/yj/YJ.java

```diff
@@ -0,0 +1,124 @@
+                            ui.print(task.toString());
+                            ui.print("Now you have like this many tasks left: " + tasks.getNumberofTasks());
+                        }
+                    } catch (IndexOutOfBoundsException e) {
+                        ui.print("Crapadoodle! You need to specify a task number or a valid task to delete.");
+                    }
+                    break;
+                case MARK: {
+                    Integer taskNumber = Parser.parseMarkCommand(input);
+                    if (taskNumber != null && tasks.getTask(taskNumber - 1) != null) {
+                        Task task = tasks.getTask(taskNumber - 1);
+                        task.markAsDone();
+                        ui.print("Niceoooo you're done wit this: ");
+                        ui.print(task.toString());
+                    }
+                    break;
+                }
+                case UNMARK: {
+                    Integer taskNumber = Parser.parseUnMarkCommand(input);
+                    if (taskNumber != null && tasks.getTask(taskNumber - 1) != null) {
+                        Task task = tasks.getTask(taskNumber - 1);
+                        task.markAsNotDone();
+                        ui.print("Ok lah you haven't finish this yet");
+                        ui.print(task.toString());
+                    }
+                    break;
+                }
+                case TODO:
+                    try {
+                        String description = Parser.parseToDoCommand(input);
+                        ToDo newToDo = new ToDo(description);
+                        tasks.addTask(newToDo);
+                        ui.print("Ok! I've added this todo! " + newToDo.toString());
+                        ui.print("You now have this many tasks: " + tasks.getNumberofTasks());
+                    } catch (IllegalArgumentException e) {
+                        ui.print(e.getMessage());
+                    }
+                    break;
+                case DEADLINE:
+                    try {
+                        Map<String, String> result = Parser.parseDeadlineCommand(input);
+                        Deadline newDeadline = new Deadline(result.get("description"), result.get("by"));
+                        tasks.addTask(newDeadline);
+                        ui.print("Ok! I've added this deadline!" + newDeadline.toString());
+                        ui.print("You now have this many tasks: " + tasks.getNumberofTasks());
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        ui.print("Crapadoodle! You need to specify a deadline in the correct format!");
+                    }
+                    break;
+                case EVENT:
+                    try {
+                        Map<String, String> result = Parser.parseEventCommand(input);
+                        Event newEvent = new Event(result.get("description"), result.get("from"), result.get("to"));
```

Comment:
Kind of dangerous to be using string keys here, maybe code in as variables for the Event class and implement getters and setters to throw compile time errors if any variable name is changed later on?